src/core/vectortile/qgsvectortilewriter.h

```diff
@@ -168,8 +174,13 @@ class CORE_EXPORT QgsVectorTileWriter
   private:
     bool writeTileFileXYZ( const QString &sourcePath, QgsTileXYZ tileID, const QgsTileMatrix &tileMatrix, const QByteArray &tileData );
     QString mbtilesJsonSchema();
+    QgsTileMatrix tileMatrixBuilder( const int &zoomLevel );
 
   private:
+    QString mDefaultCrs = QStringLiteral( "EPSG:3857" );
+    QgsCoordinateReferenceSystem mCrs;
+    QgsPointXY mTopLeft;
+    double mScaleDenom0;
```

Comment:
Having these member variables in the tile writer seems unnecessary if we have the QgsTileMatrix class - why not just have member variable e.g. `QgsTileMatrix mRootTileMatrix` which has all the information (CRS, top-left point, scale).

Also, instead of adding `setCrs()` method (which has some logic that I am not entirely sure I understand), there should be IMHO `setRootTileMatrix( const QgsTileMatrix& tm )` call that would configure the `mRootTileMatrix` member.